containers/src/Data/IntMap/Internal.hs

```diff
@@ -62,125 +299,87 @@
 -- must be checked for increased allocation when creating and modifying such
 -- functions.
 
+module Data.IntMap.Internal (
+    -- * Map Types
+      IntMap(..)
+    , L, R
+    , Node(..)
+    , IntMap_(..)
+    , DeleteResult(..)
```

Comment:
It's a bit surprising to find this so close to the main types – maybe move it closer to `delete`?

Reply:
This is a bit of a forward-looking placement. `DeleteResult` is basically `NonEmptyIntMap`, and I was kind of expecting to eventually expose it à la #608, especially since this code is already structured that way (not in this PR, of course; that would double the API surface and require a decision).

Reply:
@gereeter, if it's just a nonempty `IntMap` or `IntMap_` or similar, could you give it a name that reflects that? That would be much easier to understand.